newa/cli.py

```diff
@@ -206,75 +206,153 @@ def cmd_event(ctx: CLIContext, errata_ids: list[str]) -> None:
 
-            if action.parent_id and action.parent_id not in known_issues:
-                print(f'     !! Parent issue, "{action.parent_id}", is unknown, will try later')
-                print()
+                endless_loop_check[action.id] = queue_length
+                ctx.logger.info(f"Skipped for now (parent {action.parent_id} not yet found)")
 
                 issue_actions.append(action)
                 continue
 
-            print(f'     Issue would be assigned to {action.assignee}.')
-            print(f'       rendered: >>{render_template(action.assignee, ERRATUM=erratum_job)}<<')
-            print(f'     Will remember the issue as `{action.id}`.')
-            if action.parent_id:
-                print(f'     Issue would have issue `{action.parent_id}` as its parent.')
-            print()
-
-            known_issues[action.id] = True
-
-            # create a fake Issue object for now
-            issue = Issue(id=f'NEWA-{next(jira_id_gen)}')
-
-            if action.job_recipe:
-                print(
-                    f'* Would kick automated job for issue {action.type.name}'
-                    f'based on recipe from {action.job_recipe}:')
-                print()
-
-                jira_job = JiraJob(event=erratum_job.event,
-                                   erratum=erratum_job.erratum,
-                                   jira=issue,
-                                   recipe=Recipe(url=action.job_recipe))
-                ctx.save_jira_job('jira-', jira_job)
+            # Find existing issues related to erratum_job and action
+            search_result = jira.get_open_issues(action, all_respins=True)
+
+            # Issues related to the curent respin and previous one(s).
+            new_issues: list[Issue] = []
+            old_issues: list[Issue] = []
+            for jira_issue_key, jira_issue in search_result.items():
+                ctx.logger.info(f"Checking {jira_issue_key}")
+
+                # In general, issue is new (relevant to the current respin) if it has newa_id
+                # of this action in the description. Otherwise, it is old (relevant to the
+                # previous respins).
+                #
+                # However, it might happen that we encounter subtask issue that is new but its
+                # original parent task got dropped (by human mistake, newa would never do that).
+                # By this time new parent task already exists. Unfortunately, Jira REST API does
+                # not allow updating 'parent' field [1] and hence we cannot re-use the issue with
+                # updated parent - we need to handle it as an old one (unless it has KEEP on_respin
+                # action it will get dropped and new one is created with the proper parent).
+                #
+                # [1] https://jira.atlassian.com/browse/JRASERVER-68763
+                decision_result = ""
```

Comment:
This variable looks more like a `is_new`, a boolean flag. Is it necessary to express the new/old state with strings? May there be more states/decisions?

Reply:
Fixed (also - that condition block was made simpler).